components/player-view.js

```diff
@@ -1,318 +1,137 @@
-
-function formatTime(seconds) {
-  const secondsInDay = 24 * 60 * 60;
-  const days = Math.floor(seconds / secondsInDay);
-  seconds = seconds % secondsInDay;
-  const utc = new Date(seconds * 1000);
-  return `${days}:${utc.toLocaleTimeString('en-US', {hour12: false, timeZone: 'UTC'})}`
-    .replace(/^0:/, '') // Strip leading "0:" if days is empty.
-    .replace(/^00:/, '') // Strip leading "00:" if hours is empty.
-    .replace(/^0/, '');  // Strip leading "0" in minutes, if any.
-}
-
-module.exports = React.createClass({
-  getInitialState: function() {
-    return {showVolume: false, hovered: false};
-  },
-  step: function() {
-    const currentTime = this.isYt ? ytCtrl.getTime() : this.refs.video.currentTime;
-
-    window.AppData = deepAssign(window.AppData, {
-      currentTime: `${formatTime(currentTime)} / ${formatTime(window.AppData.duration)}`,
-      progress: currentTime / window.AppData.duration
-    });
-
-    if (currentTime >= window.AppData.duration) {
-      window.AppData = deepAssign(window.AppData, {
-        playing: false
-      });
-
-      sendMetricsEvent('player_view', 'video_ended');
-    }
-
-    if (window.AppData.playing) requestAnimationFrame(this.step);
-  },
-  onLoaded: function() {
+  onLoaded(duration) {
     clearTimeout(this.loadingTimeout);
-    sendMetricsEvent('player_view', 'video_loaded');
-    const duration = this.isYt ? ytCtrl.getDuration() : this.refs.video.duration;
 
     // for YouTube we need to detect if the duration is 0 to see
     // if there was a problem loading.
-    if (this.isYt && duration === 0) {
-      window.AppData = deepAssign(window.AppData, {error: true});
-    }
+    if ((this.props.player === 'youtube') && duration === 0) this.onError();
 
     // here we store the muted prop before it gets set in the
     // setVolume call so we can restore it afterwards.
     const wasMuted = this.props.muted;
 
     // set initial volume
-    this.setVolume({
-      target: {
-        value: this.props.volume
-      }
+    emitter.emit('set-volume', {
+      value: this.props.volume
     });
 
     // set muted/unmuted (must be called before setVolume below)
     if (wasMuted) {
-      this.mute();
+      emitter.emit('mute')
     } else {
-      this.unmute();
+      emitter.emit('unmute')
     }
 
+    emitter.emit('load', {value: duration});
+  }
 
-    window.AppData = deepAssign(window.AppData, {
-      loaded: true,
-      duration: duration
-    });
-
-    requestAnimationFrame(this.step);
-  },
-  componentWillUpdate: function() {
-    this.isYt = !!~this.props.domain.indexOf('youtube.com');
-  },
-  componentDidUpdate: function(prevProps, prevState) {
+  componentDidUpdate(prevProps, prevState) {
     if (prevProps.src !== this.props.src) {
-      this.attachVideoListeners();
+      clearTimeout(this.loadingTimeout);
+      this.loadingTimeout = setTimeout(() => {
+        deepAssign(window.AppData, {
+          error: true
+        });
+      }, 30000); // 30 second timeout
```

Comment:
I wonder if some kind of intermediate "working on it..." state should be shown after 5-10 seconds. 30 seconds is a really long time to wait on the web